homeassistant/components/kdeconnect/binary_sensor.py

```diff
@@ -0,0 +1,139 @@
+"""Support for KDE Connect binary sensors."""
+from typing import cast
+
+from pykdeconnect.client import KdeConnectClient
+from pykdeconnect.devices import KdeConnectDevice
+from pykdeconnect.plugin_registry import PluginRegistry
+from pykdeconnect.plugins.battery import BatteryReceiverPlugin
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import STATE_ON
+from homeassistant.core import HomeAssistant, State
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import KdeConnectEntity
+from .const import DATA_KEY_CLIENT, DATA_KEY_DEVICES, DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the KDE Connect binary sensors."""
+    client = cast(KdeConnectClient, hass.data[DOMAIN][DATA_KEY_CLIENT])
+    device = cast(KdeConnectDevice, hass.data[DOMAIN][DATA_KEY_DEVICES][entry.entry_id])
+
+    async_add_entities(
+        [
+            KdeConnectBatteryChargingSensor(device, client.plugin_registry),
+            KdeConnectBatteryLowSensor(device, client.plugin_registry),
+            KdeConnectConnectedSensor(device),
```

Comment:
These look similar. Can they be normalized to a single class? See `BinarySensorEntityDescription`

Reply:
`KdeConnectBatteryChargingSensor` and `KdeConnectBatteryLowSensor`(and future sensors) would be able to share the plugin loading. I would probably still need different subclasses though, because they register different callbacks on those plugins.
I don't see a way around this other than registering the callback using `getattr` with the function name as string, which doesn't seem clean.

KdeConnectConnectedSensor also can't be combined with the others, at it doesn't even use a kde connect plugin, but registers callbacks directly with the device.

Reply:
For `BinarySensorEntityDescription` you may be able use the `key` attribute to do anything specific to each sensor.

Reply:
I have now combined the `KdeConnectBatteryChargingSensor` and `KdeConnectBatteryLowSensor`, while leaving the `KdeConnectConnectedSensor` separate.

The first two use callbacks with the same signature, so I can use one callback for both.
`KdeConnectConnectedSensor` uses different callbacks. I can't use lambdas for the callbacks, as they have to be async.

Reply:
Marking this as resolved, as I've removed the binary sensor platform from this PR and will open a new PR, once the sensor platform is merged.